frontend/sw.ts

```diff
@@ -0,0 +1,154 @@
+/// <reference lib="webworker" />
+import { RouteHandler } from 'workbox-core'
+import { createHandlerBoundToURL, precacheAndRoute } from 'workbox-precaching'
+import { NavigationRoute, registerRoute, setDefaultHandler } from 'workbox-routing'
+import { NetworkFirst, NetworkOnly, StaleWhileRevalidate } from 'workbox-strategies'
+
+declare var self: ServiceWorkerGlobalScope
+export default {}
+precacheAndRoute(self.__WB_MANIFEST)
+
+self.addEventListener('install', (event) => {
+  event.waitUntil(
+    (async () => {
+      self.skipWaiting() // triggert direkt 'activate'
+    })()
+  )
+})
+
+self.addEventListener('activate', (event) => {
+  event.waitUntil(
+    (async () => {
+      self.clients.claim() // weiß nicht was genau das für einen unterschied macht
+    })()
+  )
+})
+
+// Ein benutzerdefiniertes Event für die Client-Seiten senden, wenn ein Update verfügbar ist
+self.addEventListener('message', (event) => {
+  if (event.data && event.data.type === 'SKIP_WAITING') {
+    self.skipWaiting()
+  }
+})
+
+setDefaultHandler(new NetworkOnly())
+
+// to allow work offline
+registerRoute(new NavigationRoute(createHandlerBoundToURL('index.html')))
+
+registerRoute(
+  ({ request }) => request.destination === 'font',
+  new NetworkFirst({
+    cacheName: 'font-cache'
+  })
+)
+
+//Callback Handler selbst geschrieben:
+const DataToStore: RouteHandler = async ({ request }) => {
+  let response = getDataFromStore(request)
+  return response
+}
+registerRoute(({ request }) => /\/backend\/(?!auth\b).*/.test(request.url), DataToStore)
```

Comment:
nicht in jeder umgebung haben haben die backend routen `/backend` am Ende. Das kann ja frei über die `.env` eingestellt werden. Da musst du einen anderen Ausdruck zum Prüfen nehmen.

Reply:
yes, der Gedanke kam mir auch in der Zwischenzeit mal.